app/bot/llm.py

```diff
@@ -31,17 +32,22 @@ async def generate_response(
             f"User prompt:\n{message.content}"
         )
 
+        messages: list[dict[str, str]] = [
```

Comment:
Hmm. Well, it's not the most efficient, but this does let you set the system prompt dynamically.

Reply:
What would a more efficient way be? This is the exact same format that the api needs, so we would have to convert it to this eventually

Reply:
You're recalculating the message array. Could store the whole thing rather than concatenating the system prompt every time. Not a big deal though